### PR TITLE
chore: remove macOS 10.9 specific code

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Follow up to https://github.com/electron/electron/pull/15357.
Bump the macOS deployment target to 10.10 and remove macOS 10.9 specific code.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: macOS deployment target bumped to 10.10.

### CODE REVIEW

**Comment by miniak** [comment]
@MarshallOfSound can you please review?

File: atom/browser/ui/cocoa/atom_ns_window_delegate.mm [hunk 1]
```diff
@@ -178,61 +178,55 @@ - (void)windowWillEnterFullScreen:(NSNotification*)notification {
...
-    NSWindow* window = shell_->GetNativeWindow();
-    if ((shell_->transparent() || !shell_->has_frame()) &&
-        // FIXME(zcbenz): Showing titlebar for hiddenInset window is weird under
-        // fullscreen mode.
-        // Show title if fullscreen_window_title flag is set
-        (shell_->title_bar_style() != atom::NativeWindowMac::HIDDEN_INSET ||
-         shell_->fullscreen_window_title())) {
-      [window setTitleVisibility:NSWindowTitleVisible];
-    }
+  // For frameless window we don't show set title for normal mode since the
```
Comment (MarshallOfSound) [inline atom/browser/ui/cocoa/atom_ns_window_delegate.mm]: I'm concerned that this diff still shows with `w=1` (ignore whitespace changes).

Can you verify that the line endings and tabs vs spaces haven't changed in this section?

Comment (miniak) [reply atom/browser/ui/cocoa/atom_ns_window_delegate.mm]: @MarshallOfSound SourceTree shows this properly. I did check it and I am not changing newlines and/or indentation (tabs vs. spaces)

**Comment by miniak** [comment]
@MarshallOfSound if we are not comfortable with changing Chromium’s deployment target, then I will just close this PR and re-open again when we update to a version, which does change it to 10.10

There are a few changes that can be done without bumping the deployment target, but I think all the cleanup should probably be done together.

**Pull request closed without merging**